### PR TITLE
cpu/nrf5x: ported nrfmin driver to netdev2

### PR DESCRIPTION
**Pull request by haukepetersen**
This PR
- makes nrfmin now to use netdev2
- generalizes the nrfmin driver to also work with the nrf52
- enables networking capabilities for nrf based boards by default

In the current state, `nrfmin` does however not use 6LoWPAN. It uses plain IPv6 over `nrfmin`, so it will not handle packets > 200 byte as not fragmentation is in place. But working on it...

### CODE REVIEW

File: boards/nrf52dk/Makefile.dep [hunk 1]
```diff
@@ -2,6 +2,12 @@ ifneq (,$(filter saul_default,$(USEMODULE)))
   USEMODULE += saul_gpio
 endif
 
-ifneq (,$(filter gnrc_netdev_default,$(USEMODULE)))
-  USEPKG += nordic_softdevice_ble
+ifneq (,$(USEBLE))
```
Comment (kaspar030) [inline boards/nrf52dk/Makefile.dep]: So now the default interface for nrf52dk is nrfmin?

Comment (haukepetersen) [inline boards/nrf52dk/Makefile.dep]: Ups, this was for my testing...

Will change this back to BLE, as I think that's the better default, right?

**Comment by kaspar030** [comment]
> Will change this back to BLE, as I think that's the better default, right?

I'd say yes, as probably many people using nrf52dk come for it's BLE
support.

**Comment by haukepetersen** [comment]
done

File: cpu/nrf5x_common/include/nrfmin.h [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+ * Copyright (C) 2015-2016 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_nrf5x_nrfmin NRF Minimal Radio Driver
+ * @ingroup     drivers_netdev
```
Comment (PeterKietzmann) [inline cpu/nrf5x_common/include/nrfmin.h]: `drivers_netdev_netdev2`

File: cpu/nrf5x_common/radio/nrfmin/nrfmin.c [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+{
+    iid[0] = 0;
+    iid[1] = 0xff00;
+    iid[2] = 0x00fe;
+    iid[3] = my_addr;
+}
+
+uint16_t nrfmin_get_channel(void)
+{
+    return (uint16_t)(NRF_RADIO->FREQUENCY >> 2);
```
Comment (PeterKietzmann) [inline cpu/nrf5x_common/radio/nrfmin/nrfmin.c]: Really? Old implementation did
 ```
val[0] = (0x3f & NRF_RADIO->FREQUENCY);
val[1] = 0;
```

File: cpu/nrf5x_common/radio/nrfmin/nrfmin_gnrc.c [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    pkt_snip->type = nrfmin->proto;
+
+    /* finally: remove the nrfmin header and append the netif header */
+    gnrc_pktbuf_remove_snip(pkt_snip, hdr_snip);
+    LL_APPEND(pkt_snip, netif_snip);
+
+    return pkt_snip;
+}
+
+void nrfmin_gnrc_init(void)
```
Comment (PeterKietzmann) [inline cpu/nrf5x_common/radio/nrfmin/nrfmin_gnrc.c]: This applies to the whole PR: Generally, `GNRC`-related modules/folders/functions are **pre**fixed with 'gnrc_'. Did you change the order on purpose? For me, this form was less intuitive.

File: cpu/nrf5x_common/radio/nrfmin/nrfmin.c (hunk 3 above)
Comment (haukepetersen) [inline cpu/nrf5x_common/radio/nrfmin/nrfmin.c]: Yes: this is wanted: I changed the channel handling from the previous version, now i mapped the possible frequencies to 32 channels with 4MHz spacing...

File: cpu/nrf5x_common/radio/nrfmin/nrfmin_gnrc.c (hunk 4 above)
Comment (haukepetersen) [inline cpu/nrf5x_common/radio/nrfmin/nrfmin_gnrc.c]: hm, debatable: I say this function belongs foremost to the driver, and only secondly to the GNRC stack -> hence the order. In general I would say, that driver code is not part of any network stack, and this I would say is driver code...

File: cpu/nrf5x_common/include/nrfmin.h (hunk 2 above)
Comment (haukepetersen) [inline cpu/nrf5x_common/include/nrfmin.h]: fixed

**Comment by haukepetersen** [comment]
> I wasn't able to run on the nrf52dk. The soft-device has been downloaded (as intended), flashing succeeded but there was no shell. Any hints?

Oh, seems to me that you are building the board with the wrong network driver (BLE instead of nrfmin). Just build with `BOARD=nrf52dk USEMODULE=nrfmin make ...`, then it should build using the nrfmin driver instead of the soft-device+BLE.

File: cpu/nrf5x_common/radio/nrfmin/nrfmin_gnrc.c (hunk 4 above)
Comment (kaspar030) [inline cpu/nrf5x_common/radio/nrfmin/nrfmin_gnrc.c]: hm, I think I'm with @PeterKietzmann here, for consistency reasons. All the adaption layers have their functions prefixed with gnrc_:

netdev2_eth -> gnrc_netdev2_eth
netdev2_ieee802154 -> gnrc_netdev2_ieee802154

The code for gnrc_cc110x is located in ```drivers/cc110x/gnrc_cc110x/```.

**Comment by PeterKietzmann** [comment]
Now also tested on the nrf52dk. Basic `ifconfig set` commands do work. Also checked data transmission  against nrf51dongle on different channels and with different addresses. Everything worked great. Take this as a half ACK. I don't find time (now) to look at the code again. Maybe someone else?

File: cpu/nrf5x_common/include/nrfmin.h [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+ * This driver uses the nRF5x radio in a proprietary/custom way, defining our
+ * own custom link layer. This custom link layer resembles some characteristics
+ * of the IEEE802.15.4 link layer, but is not at all compatible to it.
+ *
+ * One key point is, that this custom link layer is only meant to operate
+ * between nRF5x devices, which let's us make some very nice assumptions:
+ *  - all communicating hosts are little-endian
+ *    -> we define host byte order := network byte order
+ *
+ * The driver is using a Nordic proprietary physical layer, configured to for a
```
Comment (LudwigKnuepfer) [inline cpu/nrf5x_common/include/nrfmin.h]: `to for` seems to prose broken grammar

File: cpu/nrf5x_common/include/nrfmin.h [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+ *   ------ | ------------- | ------------- | ----- | -------------
+ *   length |   src_addr    |   dst_addr    | proto |   payload...
+ *
+ * With:
+ * - length: length of the packet, including the header -> payload len + 6
+ * - src_addr: 16-bit source address
+ * - dst_addr: 16-bit destination address
+ * - proto: type of data transferred (similar to an Ethertype field)
+ *
+ * SUMMERY:
```
Comment (LudwigKnuepfer) [inline cpu/nrf5x_common/include/nrfmin.h]: `SUMMARY`

**Comment by PeterKietzmann** [comment]
@haukepetersen I'm fine with this PR. Please address @LudwigKnuepfer's comments, rebase and squash

**Comment by haukepetersen** [comment]
fixed and squashed.

**Comment by PeterKietzmann** [comment]
`boards/microbit/Makefile.dep:13: new blank line at EOF. ERROR: This change introduces new whitespace errors`

**Comment by haukepetersen** [comment]
argh... fixed and squashed

**Comment by LudwigKnuepfer** [comment]
@PeterKietzmann I have no idea what you're talking about. GitHub points to things that don't exist anymore and the comments I still see have apparently been addressed.

**Comment by LudwigKnuepfer** [comment]
Why did you dismiss the review?

**Comment by PeterKietzmann** [comment]
I dismissed your review cause it requested changes that have been addressed. I gave green-approve-lights to merge. D'accord?

**Comment by LudwigKnuepfer** [comment]
You just did not want to wait for my approval?

**Comment by PeterKietzmann** [comment]
Yes that was the reason. And as your comments were addressed I assumed you'd reject your request for changes. Sorry Ludwig, if there are still open comments I will address those in a separate PR.

**Comment by LudwigKnuepfer** [comment]
@PeterKietzmann well I only object to this kind of process violation unless there is a good reason. I guess giving a reviewer at least on day to click the "I'm OK" button would not have caused any other problems in this case.

**Pull request merged**